### PR TITLE
[5/5] Cleanup codegen

### PR DESCRIPTION
**Pull request by paulbalaji**
#### Description

- remove old codegenjob system
- more logs from info to trace

#### Tests

- no change in generated code

#### Documentation

- [x] changelog

#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/Jobs/CodegenJob.cs [hunk 1]
```diff
@@ -16,18 +15,13 @@ public class IgnoreCodegenJobAttribute : Attribute
     public abstract class CodegenJob
     {
         public IEnumerable<string> ExpectedOutputFiles
-            => expectedOutputFiles
-                .Select(filePath => Path.Combine(OutputDirectory, filePath))
-                .Union(jobTargets.Select(target => target.FilePath));
-
-        private readonly List<string> expectedOutputFiles = new List<string>();
+            => jobTargets.Select(target => target.FilePath);
```
Comment (jamiebrynes7) [inline workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/Jobs/CodegenJob.cs]: :chefskiss:

File: workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/Model/Details/DetailsStore.cs [hunk 2]
```diff
@@ -102,27 +102,30 @@ public DetailsStore(SchemaBundle bundle, IEnumerable<string> serializationOverri
...
 
-            Logger.Info($"Populated field details of {Components.Count} components.");
+            Logger.Trace($"Populated field details of {Components.Count} components.");
 
             Logger.Trace("Removing all recursive options.");
             var numFieldsRemoved = RemoveRecursiveOptions();
-            Logger.Info($"Removed {numFieldsRemoved} recursive options.");
+            if (numFieldsRemoved > 0)
+            {
+                Logger.Info($"Removed {numFieldsRemoved} recursive options.");
```
Comment (jamiebrynes7) [inline workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/Model/Details/DetailsStore.cs]: Info -> Trace?

Comment (jamiebrynes7) [reply workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/Model/Details/DetailsStore.cs]: We get a warning elsewhere for each one right?

File: workers/unity/Packages/io.improbable.gdk.tools/GenerateCode.cs [hunk 3]
```diff
@@ -75,15 +75,14 @@ private static bool CanGenerateOnLoad()
         [MenuItem("SpatialOS/Generate code", isValidateFunction: false, MenuPriorities.GenerateCodePriority)]
         private static void GenerateMenu()
         {
-            Debug.Log("Generating code...");
```
Comment (jamiebrynes7) [inline workers/unity/Packages/io.improbable.gdk.tools/GenerateCode.cs]: Any reason for removing this?

File: workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/Model/Details/DetailsStore.cs [hunk 4]
```diff
@@ -275,7 +278,7 @@ bool IsRecursive(string fieldType, IEnumerable<string> parentTypes)
                         }
 
                         numFieldsRemoved++;
-                        Logger.Info($"Excluding field {field.CamelCaseName} from type {type.QualifiedName}.");
+                        Logger.Warn($"Excluding field {field.CamelCaseName} from type {type.QualifiedName}.");
```
Comment (zeroZshadow) [inline workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/Model/Details/DetailsStore.cs]: The warning should mention why.

File: workers/unity/Packages/io.improbable.gdk.tools/GenerateCode.cs (hunk 3 above)
Comment (paulbalaji) [reply workers/unity/Packages/io.improbable.gdk.tools/GenerateCode.cs]: woops 😅

Comment (paulbalaji) [reply workers/unity/Packages/io.improbable.gdk.tools/GenerateCode.cs]: removed it to test something and forgot to add it back

**Pull request merged**